File: 2-Basic-Mathematical-Knowledge-for-Algorithms/010_Factorial/javascript/main.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.on('line', (line) => {
    // 1. 入力処理
    const n = parseInt(line.trim());

    // 2. 階乗計算
    let result = 1;
    for (let i = 1; i <= n; i++) {
        result *= i;
    }

    // 3. 出力処理
    console.log(result);
    
    rl.close();
});

File: 2-Basic-Mathematical-Knowledge-for-Algorithms/010_Factorial/rust/src/main.rs
use std::io;

fn main() {
    // 1. 入力処理
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: u32 = input.trim().parse().expect("Please enter a number");

    // 2. 階乗計算
    let mut result: u64 = 1;
    for i in 1..=n {
        result *= i as u64;
    }

    // 3. 出力処理
    println!("{}", result);
}